docs/README.md

```diff
@@ -58,7 +84,7 @@ Note: Other versions of roxygen2 might work in SparkR documentation generation b
 To generate API docs for any language, you'll need to install these libraries:
 
 ```sh
-$ sudo pip install sphinx mkdocs numpy
+pip install sphinx==2.3.1 mkdocs==1.0.4 numpy==1.18.1
```

Comment:
@nchammas, btw why should we fix the version? I think it's better to promote the latest versions.

Reply:
We should pin dependencies for the usual reason: It guarantees a stable environment that gives reproducible results.

New releases may have breaking behavior, so you don't always want to install the latest version. And different users building the docs may have outdated versions installed, which will yield them a broken build. (e.g. If you ran `pip install sphinx` when the latest release was 1.1.0, then pip will not update your version of sphinx as new releases are made unless you explicitly request it via `pip install --upgrade`.)

These categories of build problem are addressed by pinning versions.

Reply:
I think it's OK to pin. We have to maintain this across docs vs script, but, probably better this way.

Reply:
Hm, there's tradeoff to pin the version (test and fix it compatible first vs easily breaking). I think we haven't faced any notable problem so far to maintain those implicit compatibility across versions. What about we pin the version when actually we see some overhead to maintain this?

Reply:
I'm not following your concern about pinning versions. Is there some scenario or overhead from pinning versions that you are concerned about?

It seems inevitable to me that contributors will waste time debugging issues that stem from them running different versions of these libraries. Instead of waiting for those issues to arise, it seems safe and simple enough to avoid them by pinning what we use. In fact, what prompted me to open this PR is an interaction I had with you on another PR where we were seeing different behavior when trying to build the docs. So I believe we've already seen at least one example of this problem. 😄 (I linked to that interaction from this PR's description.)

Pinning our doc build dependencies is the core idea of this PR. If we don't want to do that, then I'm happy to close this PR and leave things as-is.

Reply:
I guess I'd default to pinning the version in the absence of a strong argument either way. I do think we've had inconsistencies in the spark-website project due to differing jekyll versions, for example.

Reply:
Okie. I am good with this change.